components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/model/AuthenticationRequestUser.java

```diff
@@ -0,0 +1,168 @@
+                String claimUri = claimMap.getLocalClaim().getClaimUri();
+                if (AuthenticatorAdapterConstants.AuthRequestEntityPaths.EXTERNAL_ID_CLAIM.equals(claimUri)) {
+                    externalId = userAttributes.get(claimMap);
+                    break;
+                }
+                userClaims.add(new UserClaim(claimUri, userAttributes.get(claimMap)));
+            }
+        }
+    }
+
+    public void setIdp(String idp) {
+        this.idp = idp;
+    }
+
+    public void setSub(String sub) {
+        this.sub = sub;
+    }
+
+    public void setExternalId(String externalId) {
+        this.externalId = externalId;
+    }
+
+    public void setUserClaims(UserClaim userClaim) {
+        userClaims.add(userClaim);
+    }
+
+    /**
+     * Validate and return userType. The user type has to be either LOCAL or FED, if it is not any of them or
+     * not provided throw as exception.
+     *
+     * @return True if the idp is FED, false if idp is LOCAL.
+     * @throws ActionExecutionResponseProcessorException If invalid user type is provided.
+     */
+    public boolean resolveUserType()
+            throws ActionExecutionResponseProcessorException {
+
+        if (StringUtils.equalsIgnoreCase(
+                AuthenticatorAdapterConstants.AuthRequestEntityPaths.UserType.LOCAL.toString(), idp)) {
+            return false;
+        } else if (StringUtils.equalsIgnoreCase(
+                AuthenticatorAdapterConstants.AuthRequestEntityPaths.UserType.FED.toString(), idp)) {
+            return true;
+        } else {
+            throw new ActionExecutionResponseProcessorException("Undefined user idp type is provided for the " +
+                    "authenticated user in external authenticator service response payload: " + idp);
+        }
+    }
+
+    /**
+     * Validate and return the userId. Check if a user exists for the given userId.
+     *
+     * @return UserID for the authenticated user.
+     * @throws ActionExecutionResponseProcessorException If invalid user id is provided.
+     */
+    public String resolveUserId(AbstractUserStoreManager userStoreManager)
+            throws ActionExecutionResponseProcessorException {
+
+        if (this.getId() == null) {
+            throw new ActionExecutionResponseProcessorException("The user ID of the authenticated user in the " +
+                    "external authenticator service response payload must not be null.");
+        }
+        try {
+            if (!userStoreManager.isExistingUserWithID(this.getId())) {
+                throw new ActionExecutionResponseProcessorException("No user is found for the given user id: "
+                        +  this.getId());
+            }
+            return this.getId();
+        } catch (UserStoreException e) {
+            throw new ActionExecutionResponseProcessorException("An error occurs when trying to check whether user" +
+                    " exist for the given user id :" +  this.getId(), e );
+        }
```

Comment:
All these are input validations that needs to be incorporated for troubleshooting logs

Reply:
Fixed with commit e35699ea008a4b1df7251ac6162b6e9634aacf77